### PR TITLE
Add ref source for System.Security.Cryptography.Xml (dotnet/corefx#4278)

### PR DESCRIPTION
**Pull request by tintoy**
Hi.

Thanks to @weshaggard, I have now populated the ref source file (`ref/System.Security.Cryptography.Xml.cs`) and added the inter-project references required to compile the ref project (`ref/System.Security.Cryptography.Xml.csproj`).

This was required in order to be able to reference `System.Security.Cryptography.Xml` types in tests.

Part of #4278.

### CODE REVIEW

File: src/System.Security.Cryptography.Xml/ref/System.Security.Cryptography.Xml.csproj [hunk 1]
```diff
@@ -6,6 +6,12 @@
   </ItemGroup>
   <ItemGroup>
     <ProjectReference Include="..\..\System.Runtime\ref\System.Runtime.csproj" />
+    <ProjectReference Include="..\..\System.Runtime.Extensions\ref\System.Runtime.Extensions.csproj" />
+    <ProjectReference Include="..\..\System.Security.Cryptography.Algorithms\ref\System.Security.Cryptography.Algorithms.csproj" />
+    <ProjectReference Include="..\..\System.Security.Cryptography.Primitives\ref\System.Security.Cryptography.Primitives.csproj" />
+    <ProjectReference Include="..\..\System.Security.Cryptography.X509Certificates\ref\System.Security.Cryptography.X509Certificates.csproj" />
+    <ProjectReference Include="..\..\System.Xml.ReaderWriter\ref\System.Xml.ReaderWriter.csproj" />
+    <ProjectReference Include="..\..\System.Xml.ReaderWriter\ref\System.Xml.ReaderWriter.csproj" />
```
Comment (tintoy) [inline src/System.Security.Cryptography.Xml/ref/System.Security.Cryptography.Xml.csproj]: Duplicate removed in subsequent commit (a048a99aa9760f36a67f33c6a9ba09f17519133d).

**Comment by weshaggard** [comment]
```
EnvelopedCms\StateTests.cs(322,17): error CS0433: The type 'X509IssuerSerial' exists in both 'System.Security.Cryptography.Pkcs, Version=4.0.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' and 'System.Security.Cryptography.Xml, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' [D:\j\workspace\windows_nt_re---37265eab\src\System.Security.Cryptography.Pkcs\tests\System.Security.Cryptography.Pkcs.Tests.csproj]
```

That error is point out that the ref I generated from the src contains duplicated types from the pkcs ref. We will need to eliminate that duplication from the new library.

**Comment by tintoy** [comment]
Oops! I build the ref project but forgot to rebuild the main solution. Sorry about that. Will address it shortly.

**Comment by tintoy** [comment]
Done :)

**Comment by tintoy** [comment]
Good catch - I'll do that shortly.

**Pull request merged**